src/asynk/async_queue.rs

```diff
@@ -238,19 +252,25 @@ where
     pub async fn fetch_and_touch_task_query(
         transaction: &mut Transaction<'_>,
         task_type: &Option<String>,
-    ) -> Result<Task, AsyncQueueError> {
+    ) -> Result<Option<Task>, AsyncQueueError> {
         let task_type = match task_type {
             Some(passed_task_type) => passed_task_type,
             None => DEFAULT_TASK_TYPE,
         };
 
-        let mut task = Self::get_task_type_query(transaction, task_type).await?;
-
-        Self::update_task_state_query(transaction, &task, FangTaskState::InProgress).await?;
-
-        task.state = FangTaskState::InProgress;
-
-        Ok(task)
+        let task = match Self::get_task_type_query(transaction, task_type).await {
+            Ok(some_task) => Some(some_task),
+            Err(_) => None,
```

Comment:
Err(_) matches all types of errors. 

is it possible to match the empty result error here?

Reply:
get_task_type is gonna fail only by one thing. If get task type query fails will fail only by calling query_one method and will return tokio_postgres::error::Error ,  that's why i match every error to None.